Nouvelle page de profil (3e tentative)

Situphen opened the pull request:
Reprise de #5284 qui est elle même une reprise de #5025 (cette fois sera la bonne)

firm1 committed:
refactoring member page

firm1 committed:
fix test

firm1 committed:
ajuste l'avatar pour les grosses résolutions

firm1 committed:
remarques de la code review de gustavi

firm1 committed:
fix test

A-312 committed:
Refactoring des casquettes + effets hovers

A-312 committed:
change1

A-312 committed:
Ajout d'une variable CSS couleur

A-312 committed:
Améliore l'affichage des casquettes & informations

A-312 committed:
"HAT (zeste de savoir)" n'est plus en italique

Situphen committed:
Uniformisation

Situphen committed:
Corrige l'icône de la barre de recherche

Situphen committed:
Corrige la couleur des boutons au survol

Situphen committed:
Plusieurs modifications sur la page du profil

Situphen committed:
Nettoyage du code SCSS

Situphen committed:
Nettoyage de la nouvelle page de profil

A-312 commented:
- [x] : Le pseudo est tronqué : 
![image](https://user-images.githubusercontent.com/18501150/58442784-e4752e00-80ed-11e9-9442-4c5fb483238c.png)

 - [x] : Il manque un margin (comme pour la bibliothèque), en dessous de l'entête : 
![image](https://user-images.githubusercontent.com/18501150/58442797-f9ea5800-80ed-11e9-830f-a53010ec8cfe.png)

 - [x] : Biographie très longue, on avait dit qu'on ferait quelques choses, prochaine PR ? https://beta.zestedesavoir.com/membres/voir/Eskimon/

 - [x] : Il manque "Voir plus" quand le membre à des publications et des billets : 
![image](https://user-images.githubusercontent.com/18501150/58442861-767d3680-80ee-11e9-8f33-3260eb4e9001.png)

 - [x] : La saisie du site web (correspondant à notre MP).

A part ça tout me semble très bien !

**EDIT :** 

 - [x] : Lorsqu'un membre à beaucoup de contenu et que la description est sur 2 lignes l'alignement n'est pas le même. Est-ce gênant ?
![image](https://user-images.githubusercontent.com/18501150/58442917-e68bbc80-80ee-11e9-9cd8-0b6c9fbc7548.png)

Situphen committed:
WIP

A-312 commented:
> Lorsqu'un membre à beaucoup de contenu et que la description est sur 2 lignes l'alignement n'est pas le même. Est-ce gênant ?

On pourrait mettre un textoverflow et un `whitespace: nowrap;`.

AmauryCarrade committed:
Reprise de la page de profil

- Mise à jour du CSS & HTML des bannières bleues des pages de profil et de la
  bibliothèque, pour utiliser du flex et retirer les hacks à base de
  positionnements absolus.
- Reprise de la page de profil, mise en avant de la biographie, changement
  d'organisation de la page en deux colonnes large/étroite pour bio/stats.
- Retour du bouton d'abonnement par courriel dans un bouton combiné avec
  le nombre d'abonnés, visible également (mais non-cliquable) lorsqu'on
  voit son propre profil.
- Déplacement du badge de sanction à côté du nom
- Correction du style du badge qui était cassé sur les forums (car le même
  CSS était utilisé et qu'il a été modifié sans faire attention aux effets
  de bord).
- Déplacement du lien de signalement en bas de page, car c'est d'un usage
  assez rare.

Restant à faire

- Limiter la hauteur de la biographie, avec un lien pour agrandir en cas de
  très longue bio.
- Modifier le style des boutons d'abonnement désactivés si on clique sur un
  autre bouton (actuellement ils virent au blanc le temps du chargement, ce
  qu'est plutôt disgrâcieux).
- Mettre à jour le JS du bouton d'abonnement qui ne semble plus fonctionner
  avec le profil en bandeau.
- Supprimer complètement les blocs de dernières publications / derniers
  billets / forum s'ils sont vides (au lieu d'un message « Aucune
  publication » ou similaire).
- Peaufinements, détails, etc.

AmauryCarrade committed:
Profil : meilleure gestion des états vides

- Les profils sans aucun contenus (ni posts ni sujets ni publications ni billets)
  sont mieux présentés.
- La section “forum” des stats n'affiche plus “0 sujets” ou “0 messages postés”,
  et disparaît s'il n'y a ni l'un ni l'autre.
- Marges.

Situphen committed:
Retour en arrière (temporaire)

AmauryCarrade committed:
La biographie est tronquée si trop longue

- Ajout d'une détection de la longueur de la biographie, la tronquant si elle dépasse
  une certaine hauteur. Un “lien” est alors présenté afin de mettre en évidence le fait
  que la biographie est partielle, et de permettre de l'afficher en entier.
- Ajustements mineurs de style de code SCSS (& >).

AmauryCarrade committed:
Adaptation de la page de profil aux mobiles

- Les blocs sont réagencés, avec dans l'order vertical, casquettes, dates,
  biographie, et statistiques.
- La biographie est beaucoup plus réduite, afin de rentrer entièrement
  au dessus de la ligne de flottaison.

AmauryCarrade committed:
Changement léger du style des casquettes “staff” dans la liste

- Le « (Zeste de Savoir) » dans le nom de la casquette n'est plus blanc,
  mais bleu désaturé, et il est à peine plus petit.

AmauryCarrade committed:
Correction d'affichage de la bio si courte avec de longs secrets

- Les bios courtes mais avec de longs secrets étaient tronquées si les secrets
  étaient ouverts, sans possibilité de tout lire. Ce n'est plus le cas.

AmauryCarrade committed:
Correction des marges sur les premiers éléments de la biographie.

- La marge au dessus du premier élément des biographie était trop grande,
  lorsqu'il ne s'agissait pas d'un paragraphe.

AmauryCarrade committed:
Scripsi de ES5. Gulp beatus est.

- Conversion du code ES6 en ES5 afin que Gulp le comprenne.

artragis commented:
On en est où icin

AmauryCarrade commented:
Il en reste pas grand chose en soit (basé sur les retours qu'on a eu), l'essentiel est implémenté et fonctionne bien

- [x] Quelques personnes faisaient remarquer que la page de profil est parfois déséquilibrée visuellement quand la biographie est courte, donc il fallait que je (qu'on) réfléchisse à comment améliorer ça (par exemple, passer en une seule colonne si la biographie est courte et/ou absente).
- [x] Des détails d'alignement sont à corriger pour que le rendu soit meilleur, en particulier pour les profils sans casquettes.
- [x] Les boutons d'abonnement ne réagissent plus instantanément en JS, il faudrait restaurer ce comportement.
- [x] Il est enfin prévu de supprimer complètement les blocs de dernières publications / derniers billets / forum s'ils sont vides (au lieu d'un message « Aucune publication » ou similaire).
- [x] Et puis quelques peaufinages, mais c'est surtout ça :)

artragis commented:
Vous avez une idée de quand ça sera fini, c'est le vrai dernier élément manquant avant qu'on fasse la 28.2 et qu'on se concentre sur la 29

artragis commented:
On en est où ici

AmauryCarrade committed:
Affichage du nombre d'abonnés même quand on est déconnecté

AmauryCarrade committed:
Liens dans le résumé au dessus de la recherche

+ détails graphiques

AmauryCarrade committed:
Améliorations du bandeau

- Dates d'inscription et de connexion déplacées dans le bandeau sous la
  recherche.
- Affichage des pseudos longs amélioré (passage simple et naturel à la
  ligne)
- Le bas des lettres du pseudo passant sous la ligne de repos (par
  exemple, y, p, …) n'est plus coupé.

AmauryCarrade committed:
Meilleur bouton de signalement

AmauryCarrade committed:
Amélioration de l'intégration des casquettes et de la présence au profil.

Casquettes

- Les casquettes sont affichées dans un bloc au dessus de la biographie.
- La couleur des casquettes a été normalisée aux couleurs du site. (La teinte
  était incohérente car utilisée nulle part ailleurs.)
- Le lien de retrait des casquettes a été remplacé par une petite croix affichée
  dans le coin en haut à droite du rectangle des casquettes.
- Un lien d'explication concernant les casquettes est désormais disponible sur
  le profil.

Statistiques de présence

- Les statistiques de présence (inscription et dernière connexion) sont
  affichées :
   - sous la barre de recherche en version bureau ;
   - dans un bloc sous les casquettes en version mobile.

Reste de la page de profil

- L'indicateur d'abonnés est plus accessible (le nombre d'abonnés ne disposait
  pas de libellé pour les lecteurs d'écran).
- La géolocalisation d'un utilisateur est désormais mise en cache au niveau de
  l'instance et de l'adresse IP, afin d'éviter que la recherche dans la base de
  localisations ne soit faite quatre fois par requête vers un profil en tant que
  membre de l'équipe.

Modales

- Ajout d'un moyen de spécifier un titre indépendant du titre du lien
  ouvrant la boîte modale par le biais d'un attribut.

Autres

- Mise à jour de la documentation, ajoutant les modifications et composants
mentionnés plus haut ainsi que des rappels quant à l'accessibilité.

A-312 committed:
Ajout d'une page qui regroupe tous les contenus d'un membre

A-312 committed:
Modification du bouton abonné sur notre page de profil

A-312 committed:
Correction pour lint
Guillemet + Variable non utilisé

A-312 committed:
Corrige les icônes pour la sidebar "Voir aussi"

A-312 committed:
csslint ne râlera pas

A-312 commented:
On n'a plus accès à la liste des membres :( Je veux dire qu'on a plus de bouton car la breadcamp a disparu

A-312 commented:
Q/A : Problème avec le bouton MP quand on veut MP un membre, on s'abonne...

A-312 committed:
Corrige les tests unitaires qui ne passent pas
 - test_ban_member_is_not_contactable (zds.member.tests.tests_views.MemberTests)
 - test_hats_on_profile (zds.member.tests.tests_views.MemberTests)

A-312 commented:
Je pensais avoir corrigé les deux derniers tests. :-/

AmauryCarrade committed:
Amélioration de l'intégration des alertes.

- Les messages flash sont affichés sous le bandeau et sans marge à
  gauche.
- Les signalements sont correctement affichés, sans inversion d'ordre
  dans le texte du message (dû à un flottant malencontreux).
- Une étiquette est présente en tête de page (de la même forme que le
  badge « Banni » ou « Lecture seule ») pour signifier à l'équipe qu'un
  profil est sous le coup d'un ou plusieurs signalements.

AmauryCarrade committed:
Masquage contextuel des blocs de publication selon le contenu.

- Les blocs de publication ne sont plus affichés si vides.
- La section administrative est un peu séparée du profil public.

AmauryCarrade committed:
Alertes affichées par ordre anté-chronologique

Cohérence avec l'historique de modération

AmauryCarrade committed:
Activité et casquettes en mode vertical si pas de bio ni de signature.

- Ajout d'une classe `is-vertical` sur les linkboxes pour empiler
  verticalement les boîtes dans n'importe quel contexte.
- Retrait des CSS de la page de profil au profit de cette nouvelle
  classe, afin d'unifier le code.
- Passage des boîtes d'activité du profil ainsi que des casquettes en
  mode vertical (tout empilé, et boîtes horizontales) lorsque
  l'utilisateur n'a ni biographie, ni signature, et qu'il ne consulte
  pas sa propre page de profil (car dans ce dernier cas, un message
  l'encourageant à renseigner une biographie est affiché dans le bloc
  de cette dernière).

AmauryCarrade committed:
Abonnement en AJAX

- Adaptation du script de soumission en AJAX afin qu'il fonctionne
  également sur les boutons combinés.
- Corrections de styles mineures.

AmauryCarrade commented:
Sauf tests et éventuels peaufinages de code ou de visuel, ça devrait être bon ici. Si ce n'est qu'on a perdu le lien vers la liste des membres (il s'est dit que ça pouvait attendre une autre PR).
Aussi, la case à cocher du premier message peut l'être depuis un moment.

![Aperçu de la version actuelle de la page de profil, vu d'un compte d'équipe](https://user-images.githubusercontent.com/1417570/63216299-a9eec080-c133-11e9-8d9f-a923d9365ae3.png)

Situphen commented:
@AmauryCarrade Super nouvelle ! **J'ai mis la PR sur la bêta** et j'essaie de relire le code demain si possible. Par contre, il y a deux tests à corriger : 

```
FAIL: test_ban_member_is_not_contactable (zds.member.tests.tests_views.MemberTests)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/zestedesavoir/zds-site/zds/member/tests/tests_views.py", line 1172, in test_ban_member_is_not_contactable

    self.assertContains(result, phrase)

  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/testcases.py", line 369, in assertContains

    self.assertTrue(real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr)

AssertionError: False is not true : Couldn't find 'Envoyer un message privé' in response

======================================================================

FAIL: test_hats_on_profile (zds.member.tests.tests_views.MemberTests)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/zestedesavoir/zds-site/zds/member/tests/tests_views.py", line 1406, in test_hats_on_profile

    self.assertContains(result, _('Casquettes'))

  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/testcases.py", line 369, in assertContains

    self.assertTrue(real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr)

AssertionError: False is not true : Couldn't find 'Casquettes' in response
```

A-312 commented:
J'avais essayais mais je n'ai pas réussi : https://github.com/zestedesavoir/zds-site/pull/5350/commits/f191fd0c97a8b6e748c9b3941107e6c989aba7b4

A-312 committed:
fix test: test_ban_member_is_not_contactable

A-312 committed:
Meilleur validation du formulaire pour les sites

A-312 commented:
Je n'arrive pas à corriger les tests... parce que ce n'est pas les tests d'origine, ils ont été modifié entre temps (pour que le premier test passe mais apparemment les tests suivants n'ont pas été modifié). En corrigeant les tests suivant (d'origine donc), j'ai rendu faux le premier test (qui a été modifié).

A-312 committed:
Fix: test_hats_on_profile

A-312 commented:
```
======================================================================
ERROR: test_hats_on_profile (zds.member.tests.tests_views.MemberTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/zestedesavoir/zds-site/zds/member/tests/tests_views.py", line 1401, in test_hats_on_profile
    self.assertNotContains(result.content.decode('utf-8'), _('Casquettes'))
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/testcases.py", line 378, in assertNotContains
    response, text, status_code, msg_prefix, html)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/testcases.py", line 331, in _assert_contains
    response.status_code, status_code,
AttributeError: 'str' object has no attribute 'status_code'
----------------------------------------------------------------------
```

AmauryCarrade committed:
Mise à jour des tests

- test_ban_member_is_not_contactable
  - Renommé test_members_are_contactable
  - Corrigé car le cas des utilisateurs déconnectés était mal géré

- test_hats_on_profile
  - Retrait d'un sous-test ne faisant pas sens (les casquettes ne sont
    pas systématiquement visibles même quand il n'y en a pas pour
    l'équipe)
  - Ajout d'un sous-test assurant la présence du lien d'ajout de
    casquette

- test_add_hat
  - Correction d'un sous-test qui n'était pas terminé et concrètement,
    jamais testé : celui des casquettes contenant des caractères utf8mb4

AmauryCarrade committed:
Revert "Fix: test_hats_on_profile"

This reverts commit dc6bc62bfb2b7f7c47a547713cdb8622a3929e5f.

AmauryCarrade commented:
Ça passe en local (les tests concernant les membres), espérons aussi avec Travis

AmauryCarrade committed:
Correction de bugs d'affiche en version mobile sans bio

- Pour un utilisateur n'ayant pas de bio ni de signature, en version
  mobile, les blocs de statistiques ne prenaient pas toute la largeur,
  et un titre restait incorrectement au dessus du bloc “Site web”.
- Équilibrage des pages sans bio mais avec signature en augmentant la
  taille du bloc vide.

AmauryCarrade committed:
Cohérence de l'affichage du nombre d'abonnés.

- Quelque soit le contexte, quand le nombre d'abonnés est écrit de façon
  complète (pas juste le nombre) et qu'il est nul, il est écrit « Aucun
  abonné » et non « 0 abonné ».

Situphen commented on assets/scss/components/_user-profile.scss:
```diff
@@ -1,10 +1,329 @@
+                            rgba(#FDFDFD, 0.4) 20%,
+                            rgba(#FDFDFD, 0.6) 40%,
+                            rgba(#FDFDFD, 0.9) 80%,
+                            rgba(#FDFDFD, 1) 100%
```
On ne devrait pas avoir simplement une indentation de 4 espaces ? Ça fait un peu beaucoup là je trouve

AmauryCarrade replied:
Initialement c'était aligné avec autre chose mais cet autre chose a disparu et c'est resté
Corrigé

Situphen commented on assets/scss/pages/_flexpage.scss:
```diff
@@ -46,50 +46,164 @@ $content-width: 1145px;
-                left: 0;
-                border-bottom: 30.02px solid rgba(0, 0, 0, 0.2);
+            // position: relative;
+            // bottom: -80px;
```
Commentaires à supprimer ?

Situphen commented on assets/scss/pages/_flexpage.scss:
```diff
@@ -195,16 +324,59 @@ $content-width: 1145px;
+                    padding: .4rem 0;
+                }
+
+                ul {
```
Suggestion : 

```scss
p, ul {
    // foo
}

ul li {
    // bar
}
```

Situphen commented on doc/source/front-end/elements-specifiques-au-site.rst:
```diff
@@ -60,20 +60,47 @@ de l'attribut (ici ``Fermez-moi !``) deviendra le texte du bouton.
+
+   <div id="une-ancre" class="modal" data-modal-close="Fermez-moi !">
+       Une super boîte modale !
+        <a class="btn btn-success" href="#">Cliquez moi !</a>
```
```suggestion
        <a class="btn btn-submit" href="#">Cliquez moi !</a>
```

Situphen commented on doc/source/front-end/elements-specifiques-au-site.rst:
```diff
@@ -60,20 +60,47 @@ de l'attribut (ici ``Fermez-moi !``) deviendra le texte du bouton.
-La création du lien affichant la boîte modale est tout aussi simple : il suffit de mettre une ancre correspondant à l'``id`` de la boîte modale ainsi que la classe ``open-modal`` :
+La création du lien affichant la boîte modale est tout aussi simple : il suffit
+de mettre une ancre correspondant à l'``id`` de la boîte modale ainsi que la
+classe ``open-modal``. N'oubliez pas l'attribut ``aria-haspopup="dialog"`` pour
```
Gros +1

Situphen commented on doc/source/front-end/elements-specifiques-au-site.rst:
```diff
@@ -60,20 +60,47 @@ de l'attribut (ici ``Fermez-moi !``) deviendra le texte du bouton.
+-----------------------------------------
+
+Si vous voulez un autre titre que le texte du lien pour votre modale, vous pouvez ajouter
+l'attribut ``data-modal-title="Le titre de ma boîte"`` à le boîte modale. Le texte
```
```suggestion
l'attribut ``data-modal-title="Le titre de ma boîte"`` à la boîte modale. Le texte
```

AmauryCarrade committed:
Marges absentes entre 960 et 1140px de large

AmauryCarrade committed:
Alignement des casquettes + fond au survol

Situphen commented on templates/member/profile.html:
```diff
@@ -15,286 +20,833 @@
+                                                {% if link %}<a href="{{ link }}">{% endif %}
+                                                {{ text }}{% if link %}</a>{% endif %}{% endcaptureas %}
+
+                                                {% if forloop.last and not forloop.first %}
```
Je propose ce code, qui me parait plus simple à comprendre et devrait faire la même chose sauf erreur de ma part : 

```html+django
{% if forloop.last %}
    {{ summary_link_html }}
{% elif forloop.revcounter == 2 %}
    {{ summary_link_html }} et
{% else %}
    {{ summary_link_html }},
{% endif %}
```

AmauryCarrade replied:
En effet, bien vu, je prends ta version bien plus lisible

AmauryCarrade committed:
Intégration plus naturelle de l'IP

Situphen commented on templates/member/profile.html:
```diff
@@ -15,286 +20,833 @@
+                    {% elif not user.is_authenticated %}
+                        <div class="actions">
+                            <div class="btn btn-ico ico-after star light" disabled>
+                                {% if subscriber_count == 0 %}
```
Pourquoi pas tout simplement ça ?

```html+django
<div class="btn btn-ico ico-after star light" disabled>
    {{ subscriber_count_aria_label }}
</div>
```

AmauryCarrade committed:
Bloc de bio plus haut sur téléphone

Situphen commented on templates/member/profile.html:
```diff
@@ -15,286 +20,833 @@
+            {% if topics %}
+                <section>
+                    <h2 itemprop="name">
+                        {% trans 'Derniers sujets crées' %}
```
```suggestion
                        {% trans 'Derniers sujets créés' %}
```

Situphen commented on zds/member/models.py:
```diff
@@ -92,8 +93,12 @@ def get_city(self):
         Uses geo-localization to get physical localization of a profile through its last IP address.
         This works relatively well with IPv4 addresses (~city level), but is very imprecise with IPv6 or exotic internet
         providers.
+        The result is cached on an instance level because this method is called a lot in the profile.
```
Si je ne me trompe pas, ce n'est affiché que pour les membres Staff donc je ne suis pas sûr que ce soit vraiment nécessaire. Cela dit c'est codé et le code m'a l'air correct donc autant garder ça je pense. @artragis un avis sur le sujet ?

artragis replied:
en fait la géolocalisation c'est toujours coûteux et on n'a pas forcément beaucoup de ressources côté serveur donc si on peut éviter des calculs inutles, on prend.

AmauryCarrade replied:
(C'est effectivement que pour les membres de l'équipe, j'ai fait attention)

J'estime que ce n'est pas parce que c'est que pour l'équipe que les performances doivent être mauvaises (même si oui, c'est _moins important_) ^^

Situphen commented on zds/member/tests/tests_views.py:
```diff
@@ -1166,24 +1167,29 @@ def test_ban_member_is_not_contactable(self):
         self.assertNotContains(result, phrase)
 
+        # If an user is logged in, the PM button is shown for other normal users
         self.assertTrue(self.client.login(username=user_2.user.username, password='hostel77'))
```
Est-ce que c'est bien la peine de connecter/déconnecter trois fois *user_2* ?

AmauryCarrade replied:
Certes non, le test était initialement comme ça mais à la réflexion c'est peu logique

Situphen commented on zds/tutorialv2/urls/urls_contents.py:
```diff
@@ -19,6 +19,10 @@
     re_path(r'^flux/atom/$', RedirectView.as_view(pattern_name='publication:feed-atom',
                                                   permanent=True), name='feed-atom'),
 
+    re_path(r'^(?P<pk>\d+)/$',
```
Je ne sais pas si c'est une bonne idée d'utiliser `/contenus/<pk>/` pour afficher tous les contenus du membre dont l'identifiant est \<pk\> quand il existe déjà `/contenus/<pk>/<slug>` pour afficher le contenu dont l'identifiant est \<pk\> et le slug est \<slug\>. Je trouve qu'il y a beaucoup de risques de confusion.

AmauryCarrade replied:
Pourquoi pas `/contenus/<username>` ? En plus l'URL serait plus jolie à partager

Situphen replied:
C'est une bonne idée ! Par contre pour filtrer le type de contenu, on utilise `/contenus/<username>/tutoriels` ou `/contenus/<username>?type=tutoriel` ? Aussi, ça va casser les anciennes URLs comme `/contenus/tutoriels/<user_pk>` mais on peut mettre en place une redirection si besoin.

AmauryCarrade replied:
Je serais plus partant de ta première URL. Et de faire une redirection : je trouve plus propre de référencer les contenus de quelqu'un avec son pseudo qu'avec son numéro.

On pourrait argumenter qu'on autorise les pseudos avec des espaces et autres caractères spéciaux et que ça peut faire des URLs bizarres, mais c'est déjà le cas pour l'URL du profil donc…

Situphen replied:
Bah go alors !

AmauryCarrade committed:
Remarques de Situphen

AmauryCarrade committed:
Intégration encore plus naturelle de l'IP

noraj du perfectionnaj

AmauryCarrade committed:
Modification des URLs des listes de contenus d'un utilisateur.

Nouvelles URLs :

/contenus/voir/<username>/
/tutoriels/voir/<username>/
/articles/voir/<username>/
/billets/voir/<username>/

Les anciennes URLs avec un identifiant redirigent.

A-312 commented:
Pourquoi `voir` ? et pas `author` ?

AmauryCarrade commented:
Par cohérence avec la page de profil : `/membres/voir/<username>`

A-312 commented:
Ici username est le nom du membre par cohérence on devrait avoir : `/tuto/voir/<tutoname>/` :yum:

AmauryCarrade commented:
Rhôô :p 
Non plus sérieusement pour la liste on trouvait ça assez cohérent et propre avec @Situphen. Puis au bout d'un moment il faut bien faire un choix ^^

Situphen commented on zds/tutorialv2/views/contents.py:
```diff
@@ -1958,3 +1958,27 @@ def get_context_data(self, **kwargs):
+
+class RedirectOldContentOfAuthor(RedirectView):
+    """
+    allows to redirect /tutoriels/beta/old_pk/slug to /contenus/beta/new_pk/slug
```
Un oubli ? :-°

AmauryCarrade replied:
complètement

Situphen commented on zds/tutorialv2/views/contents.py:
```diff
@@ -428,7 +428,7 @@ def delete(self, request, *args, **kwargs):
             messages.success(self.request, _('Vous avez bien supprimé {}.').format(_type))
 
-        return redirect(reverse('content:find-' + object_type, args=[request.user.pk]))
+        return redirect(reverse(self.object.type.lower() + ':find-' + object_type, args=[request.user.username]))
```
`object_type` au lieu de `self.object.type.lower()`, ou sinon supprimer la variable à la rigueur, ou ne rien faire car je chipote là

AmauryCarrade replied:
Ah oui, c'est juste un copié/collé des autres cas similaires, c'est corrigé
(Mince les tests passent pas sur Travis)

Situphen commented on zds/tutorialv2/views/contents.py:
```diff
@@ -1958,3 +1958,27 @@ def get_context_data(self, **kwargs):
+    allows to redirect /tutoriels/beta/old_pk/slug to /contenus/beta/new_pk/slug
+    """
+    permanent = True
+    type = 'TUTORIAL'
```
à supprimer ?

AmauryCarrade replied:
Non, nécessaire sinon Django refuse de passer le paramètre à la vue

AmauryCarrade committed:
PEP 8 et remarques de Situphen

AmauryCarrade committed:
Oops j'avais oublié le commentaire

AmauryCarrade committed:
RedirectOldContentOfAuthor : type = None

Situphen commented:
**Revue de code** : OK
**QA :** OK
**Travis :** OK

@artragis C'est déployé sur la bêta si tu veux vérifier une dernière fois, pour moi tout est bon ! Je te laisse l'honneur de merger cette PR ; c'est ta revanche sur #4942 :P

gustavi commented on zds/tutorialv2/views/contents.py:
```diff
@@ -1948,3 +1960,28 @@ def get_context_data(self, **kwargs):
+            route = 'opinion:find-opinion'
+
+        if not route:
+            raise Http404('Ce type de contenu est inconnu dans le système')
```
Au risque de faire le pénible, manque pas de l'i18n ici ? :angel:

artragis replied:
Le message ne s'affiche qu'au débug non?

gustavi replied:
Tu me mets un vrai doute. Au pire c'est pas grave, ça peut être corrigé plus tard, il doit manquer de l'i18n à d'autres endroits.

artragis replied:
Parfait alors. (je dis ça car personnellement dans le module de tuto, je n'ai mis ces mesages  que pour débugger les types de 404 qu'on y lance, côté prod on n'a que notre page unique "page non trouvée".
